src/network/NetworkManagerSecondary.hh

```diff
@@ -81,6 +92,24 @@ namespace ignition
 
       /// \brief Collection of performers associated with this secondary.
       private: std::unordered_set<Entity> performers;
+
+      /// \brief Thread doing steps asynchronously.
+      private: std::thread steppingThread;
+
+      /// \brief Vector of received step messages.
+      private: std::deque<private_msgs::SimulationStep> steps;
+
+      /// \brief Mutex protecting `this->steps`.
+      private: std::mutex stepsMutex;
+
+      /// \brief Condition variable used to awake the thread doing steps.
+      private: std::condition_variable moreStepsCv;
+
+      /// \brief Boolean used to indicate that the thread
+      /// doing asynchronous steps has to stop.
+      private: bool stopAsyncStepThread {false};
```

Comment:
Should this be an atomic?

Reply:
Totally right, this caused segfaults. Addressed with: a5aae74